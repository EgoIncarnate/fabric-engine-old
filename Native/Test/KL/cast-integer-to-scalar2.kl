operator entry() 
{ 
  var Integer nb_points = 3;
  for( var Integer i = 0; i < nb_points; i++ )
    report( Scalar( i ) / Scalar( nb_points - 1 ) );
      
} 
