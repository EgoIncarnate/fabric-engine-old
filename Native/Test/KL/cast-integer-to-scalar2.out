GlobalList
  Function( 'entry' ('entry'), void )
    ParamList
    CompoundStatement
      StatementList
        StatementList
          AssignedVarDecl( VarDecl( 'nb_points', Integer ), ConstSize( 3 ) )
            ConstSize( 3 )
        CStyleLoop
          AssignedVarDecl( VarDecl( 'i', Integer ), ConstSize( 0 ) )
            ConstSize( 0 )
          BinOp( LT )
            Var('i')
            Var('nb_points')
          UniOp( postfix ++ )
            Var('i')
          Report
            BinOp( DIV )
              TempValue( Scalar )
                ArgList
                  Var('i')
              TempValue( Scalar )
                ArgList
                  BinOp( SUB )
                    Var('nb_points')
                    ConstSize( 1 )
0
0.5
1
